src/openrct2/localisation/Formatting.cpp

```diff
@@ -0,0 +1,207 @@
+        {
+            ss << buffer[j];
+        }
+    }
+
+    template<typename T> void FormatArgument(std::stringstream& ss, FormatToken token, T arg)
+    {
+        switch (token)
+        {
+            case FORMAT_UINT16:
+            case FORMAT_INT32:
+                if constexpr (std::is_integral<T>())
+                {
+                    FormatNumber<0, false>(ss, arg);
+                }
+                break;
+            case FORMAT_COMMA16:
+            case FORMAT_COMMA32:
+                if constexpr (std::is_integral<T>())
+                {
+                    FormatNumber<0, true>(ss, arg);
+                }
+                break;
+            case FORMAT_COMMA1DP16:
+                if constexpr (std::is_integral<T>())
+                {
+                    FormatNumber<1, true>(ss, arg);
+                }
+                else if constexpr (std::is_floating_point<T>())
+                {
+                    FormatNumber<1, true>(ss, std::round(arg * 10));
+                }
+                break;
+            case FORMAT_COMMA2DP32:
+                if constexpr (std::is_integral<T>())
+                {
+                    FormatNumber<2, true>(ss, arg);
+                }
+                else if constexpr (std::is_floating_point<T>())
+                {
+                    FormatNumber<2, true>(ss, std::round(arg * 100));
+                }
+                break;
+            case FORMAT_VELOCITY:
+                if constexpr (std::is_integral<T>())
+                {
+                    switch (gConfigGeneral.measurement_format)
+                    {
+                        default:
+                        case MeasurementFormat::Imperial:
+                            FormatStringId(ss, STR_UNIT_SUFFIX_MILES_PER_HOUR, arg);
+                            break;
+                        case MeasurementFormat::Metric:
+                            FormatStringId(ss, STR_UNIT_SUFFIX_KILOMETRES_PER_HOUR, mph_to_kmph(arg));
+                            break;
+                        case MeasurementFormat::SI:
+                            FormatStringId(ss, STR_UNIT_SUFFIX_METRES_PER_SECOND, mph_to_dmps(arg));
+                            break;
+                    }
+                }
+                break;
+            case FORMAT_STRING:
+                if constexpr (std::is_same<T, const char*>())
+                {
+                    ss << arg;
+                }
+                else if constexpr (std::is_same<T, const std::string&>())
+                {
+                    ss << arg.c_str();
+                }
+                break;
+            case FORMAT_STRINGID:
+            case FORMAT_STRINGID2:
+                if constexpr (std::is_integral<T>())
+                {
+                    ss << language_get_string(arg);
+                }
+                break;
+        }
+    }
+
+    std::pair<std::string_view, uint32_t> FormatNextPart(std::string_view& fmt)
```

Comment:
I find the whole idea of changing `fmt` on this function a bit weird, can you elaborate, please?

Reply:
This no longer exists.